qcodes/instrument_drivers/Spectrum/M4i.py

```diff
@@ -864,6 +892,34 @@ def gated_trigger_acquisition(self, mV_range, memsize, pretrigger_size, posttrig
 
         return voltages
 
+    def single_software_trigger_acquisition_boxcar(self, mV_range, memsize, posttrigger_size):
+        """ Acquire a single data trace with boxcar averaging
+
+        Args:
+            mV_range (float): range in mV
+            memsize (int): size of data trace
+            posttrigger_size (int): size of data trace after triggering
+        Returns:
+            voltages (array)
+        """
+        self.card_mode(pyspcm.SPC_REC_STD_BOXCAR)  # single
+
+        self.segment_size(memsize)
+        self.posttrigger_memory_size(posttrigger_size)
+        self.data_memory_size(memsize * self.box_averages())
+        numch = self._num_channels()
+
+        self.trigger_or_mask(pyspcm.SPC_TMASK_SOFTWARE)
+        self.general_command(pyspcm.M2CMD_CARD_START |
+                             pyspcm.M2CMD_CARD_ENABLETRIGGER | pyspcm.M2CMD_CARD_WAITREADY)
+
+        output = self._transfer_buffer_numpy(memsize, numch, bytes_per_sample=4)

```

Comment:
should bytes_per_sample also be an argument of the single_software_trigger_acquisition_boxcar method? because now there is no way to run single_software_trigger_acquisition_boxcar for bytes_per_sample==2.

Reply:
The boxcar acquisition only uses the int32 datatype. (this is required since the boxcar method adds values, so the int16 would overflow)